bnr-utils/nzsyncbackup/main.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+    "flag"
+    "log"
+    "nzconnector/connector"
+    "nzconnector/factory"
+)
+
+const(
+    upload = iota
+    download
+)
+
+func parseConnectorArgs(e Connector.IConnector, args string) {
+    if e != nil {
+        e.ParseConnectorArgs(args)
+    }
+}
+
+
+func main() {
+    var backupinfo Connector.BackupInfo
+    var connectorInfo Connector.ConnectorInfo
+    var otherargs Connector.OtherArgs
+    var err error
+// parse input args
+    Connector.ParseArgs(&backupinfo, &connectorInfo, &otherargs)
+    flag.Parse()
+    Connector.SetUpLogFile(&backupinfo, &connectorInfo, &otherargs)
+    connector := Factory.GetConnector(connectorInfo.Connector)
+    parseConnectorArgs(connector, connectorInfo.ConnectorArgs)
```

Comment:
Lets use [Cobra](https://github.com/spf13/cobra) for parsing here. 

Lets make this 

```
nzsyncbackup [upload|download|list] [aws-s3|azure-blob|ibm-cos]

# where connector args would be
# for aws s3 
nzsyncbackup upload aws-s3 -h

Options
-------

-access-key <ACCESS-KEY>      The access key for the object store [AWS_ACCESS_KEY_ID] (required)
-bucket-name <BUCKET-NAME>    The name of the bucket to store backups to (required)
-region <REGION>              The region of the object store bucket (required)
-secret-key <SECRET-KEY>      The secret key for the object store [AWS_SECRET_ACCESS_KEY] (required)
--unique-id <id>

# .. similar options for ibm-cos
# for azure 

nzsyncbackup upload azure-blob -h

Options
-------

-account-key <ACCOUNT-KEY>      The Azure Blob account key (required)
-account-name <ACCOUNT-NAME>    The Azure Blob account name (required)
-blob-type <BLOB-TYPE>          The type of Azure Blob container. One of [block,page]
-bucket-name <BUCKET-NAME>      The name of the bucket to store backups to (required)
-container <CONTAINER>          The Azure Blob container name (required)
```

ditto for download and list